db/db.go

```diff
@@ -20,8 +22,34 @@ const (
 	DeduplicatorCache
 	// MailserversCache is a list of mail servers provided by users.
 	MailserversCache
+	// TopicHistoryBucket isolated bucket for storing history metadata.
+	TopicHistoryBucket
+	// HistoryRequestBucket isolated bucket for storing list of pending requests.
+	HistoryRequestBucket
 )
 
+// NewMemoryDB returns leveldb with memory backend prefixed with a bucket.
+func NewMemoryDB() (*leveldb.DB, error) {
+	return leveldb.Open(storage.NewMemStorage(), nil)
+}
+
+// NewIsolatedDB returns instance that ensures isolated operations.
+func NewIsolatedDB(db *leveldb.DB, prefix storagePrefix) PrefixedLevelDB {
```

Comment:
isolated -> prefixed

Reply:
comments are not fixed in this branch. and this PR is based on 1433